packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1507,7 +1507,10 @@ export class SnapController extends BaseController<
     );
   }
 
-  _onUnhandledSnapError(snapId: string, _error: SnapErrorJson) {
+  _onUnhandledSnapError(snapId: string, error: SnapErrorJson) {
+    // Log the error that caused the crash
+    // so it gets raised to the developer for debugging purposes.
+    logError(error);
```

Comment:
Maybe add some context?
```suggestion
    logError(`Unhandled error from "${snapId}": ${error}`);
```